db/background_mgr_attachment_migration_test.go

```diff
@@ -0,0 +1,312 @@
+		_, doc, err := collection.Put(ctx, key, docBody)
+		require.NoError(t, err)
+		assert.NotNil(t, doc.SyncData.Attachments)
+	}
+
+	// Move some subset of the documents attachment metadata from global sync to sync data
+	for j := 0; j < 5; j++ {
+		key := fmt.Sprintf("%s_%d", t.Name(), j)
+		value, xattrs, cas, err := collection.dataStore.GetWithXattrs(ctx, key, []string{base.SyncXattrName, base.GlobalXattrName})
+		require.NoError(t, err)
+		syncXattr, ok := xattrs[base.SyncXattrName]
+		assert.True(t, ok)
+		globalXattr, ok := xattrs[base.GlobalXattrName]
+		assert.True(t, ok)
+
+		var attachs GlobalSyncData
+		err = base.JSONUnmarshal(globalXattr, &attachs)
+		require.NoError(t, err)
+
+		MoveAttachmentXattrFromGlobalToSync(t, ctx, key, cas, value, syncXattr, attachs.GlobalAttachments, true, collection.dataStore)
+	}
+
+	attachMigrationMgr := NewAttachmentMigrationManager(db.MetadataStore, db.MetadataKeys)
+	require.NotNil(t, attachMigrationMgr)
+
+	options := map[string]interface{}{
+		"database": db,
+	}
+	err := attachMigrationMgr.Start(ctx, options)
+	require.NoError(t, err)
+
+	// wait for task to complete
+	err = WaitForConditionWithOptions(t, func() bool {
+		var status BackgroundManagerStatus
+		rawStatus, _ := attachMigrationMgr.GetStatus(ctx)
+		_ = base.JSONUnmarshal(rawStatus, &status)
+		return status.State == BackgroundProcessStateCompleted
+	}, 200, 200)
+	require.NoError(t, err)
+
+	// assert that the subset (5) of the docs were changed, all created docs were processed (10)
+	stats := getAttachmentMigrationStats(attachMigrationMgr.Process)
+	assert.Equal(t, int64(10), stats.DocsProcessed)
+	assert.Equal(t, int64(5), stats.DocsChanged)
+
+	// assert that the sync info metadata version doc has been written to the database collection
+	var syncInfo base.SyncInfo
+	_, err = collection.dataStore.Get(base.SGSyncInfo, &syncInfo)
+	require.NoError(t, err)
+	assert.Equal(t, int8(4), syncInfo.MetaDataVersion)
+
+}
+
+func getAttachmentMigrationStats(resyncManager BackgroundManagerProcessI) ResyncManagerResponseDCP {
+	var resp ResyncManagerResponseDCP
+	rawStatus, _, _ := resyncManager.GetProcessStatus(BackgroundManagerStatus{})
+	_ = base.JSONUnmarshal(rawStatus, &resp)
+	return resp
+}
+
+func TestAttachmentMigrationManagerResumeStoppedMigration(t *testing.T) {
+	if base.UnitTestUrlIsWalrus() {
+		t.Skip("rosmar does not support DCP client, pending CBG-4249")
+	}
+	base.LongRunningTest(t)
+
+	db, ctx := setupTestDB(t)
+	defer db.Close(ctx)
+	collection, ctx := GetSingleDatabaseCollectionWithUser(ctx, t, db)
+
+	// create some docs with attachments defined
+	for i := 0; i < 4000; i++ {
+		docBody := Body{
+			"value":         1234,
+			BodyAttachments: map[string]interface{}{"myatt": map[string]interface{}{"content_type": "text/plain", "data": "SGVsbG8gV29ybGQh"}},
+		}
+		key := fmt.Sprintf("%s_%d", t.Name(), i)
+		_, doc, err := collection.Put(ctx, key, docBody)
+		require.NoError(t, err)
+		assert.NotNil(t, doc.SyncData.Attachments)
```

Comment:
```suggestion
		require.NotNil(t, doc.SyncData.Attachments)
```

If this error is hit, the rest of the code does not work.